CMakeLists.txt

```diff
@@ -1,340 +1,306 @@
-cmake_minimum_required(VERSION 2.8.12)
 project(obs-browser)
 
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}")
-set(CMAKE_INCLUDE_CURRENT_DIR TRUE)
-
-include_directories(SYSTEM "${CMAKE_SOURCE_DIR}/UI/obs-frontend-api")
-include_directories("${CMAKE_CURRENT_SOURCE_DIR}")
-include_directories("${CMAKE_CURRENT_SOURCE_DIR}/deps")
-
-option(BROWSER_LEGACY "Use legacy CEF version 3770" OFF)
-
-find_package(CEF QUIET)
-
-if(NOT CEF_FOUND)
-	message(FATAL_ERROR "CEF Not found -- set CEF_ROOT_DIR or set BUILD_BROWSER=OFF.")
+option(
+  ENABLE_BROWSER_SOURCE
+  "Enable building OBS with browser source plugin (required Chromium Embedded Framework)"
+  ON)
+if(NOT ENABLE_BROWSER_SOURCE OR NOT ENABLE_UI)
+  message(STATUS "OBS: obs-browser plugin not enabled")
+  return()
 endif()
 
-if(WIN32 OR (APPLE AND NOT BROWSER_LEGACY))
-    option(SHARED_TEXTURE_SUPPORT_ENABLED "Enable shared texture support for the browser plugin" ON)
-else()
-    option(SHARED_TEXTURE_SUPPORT_ENABLED "Enable shared texture support for the browser plugin" OFF)
+if(NOT DEFINED BUILD_QT)
+  set(BUILD_QT
+      5
+      CACHE STRING "Qt version used to build OBS app and UI" FORCE)
+  set_property(CACHE ENABLE_RTMPS PROPERTY STRINGS 5 6)
 endif()
 
-option(BROWSER_PANEL_SUPPORT_ENABLED "Enables Qt web browser panel support" ON)
+set(BROWSER_SOURCE_AVAILABLE
+    ON
+    CACHE BOOL "Browser source plugin available" FORCE)
 
-if(NOT APPLE)
-	option(USE_QT_LOOP "Runs CEF on the main UI thread alongside Qt instead of in its own thread" OFF)
-else()
-	option(USE_QT_LOOP "Runs CEF on the main UI thread alongside Qt instead of in its own thread" ON)
+if(OS_MACOS)
+  option(ENABLE_BROWSER_LEGACY
+         "Use legacy CEF version 3770 for browser source plugin" OFF)
 endif()
 
-if(UNIX AND NOT APPLE)
-	find_package(X11 REQUIRED)
-	if(NOT X11_FOUND)
-		message(FATAL_ERROR "X11 Not found -- install a X11 environment or set BUILD_BROWSER=OFF.")
-	endif()
-	include_directories("${X11_INCLUDE_DIR}")
+if(OS_MACOS OR OS_WINDOWS)
+  option(ENABLE_BROWSER_SHARED_TEXTURE
+         "Enable shared texture support for browser source plugin" ON)
 endif()
 
+option(ENABLE_BROWSER_PANELS "Enable Qt web browser panel support" ON)
+option(ENABLE_BROWSER_QT_LOOP
+       "Enable running CEF on the main UI thread alongside Qt" ${OS_MACOS})
 
-configure_file(
-	"${CMAKE_CURRENT_SOURCE_DIR}/browser-config.h.in"
-	"${CMAKE_CURRENT_BINARY_DIR}/browser-config.h")
-
-include_directories("${CMAKE_CURRENT_BINARY_DIR}")
-include_directories("${CEF_ROOT_DIR}")
-
-# ----------------------------------------------------------------------------
+mark_as_advanced(ENABLE_BROWSER_LEGACY ENABLE_BROWSER_SHARED_TEXTURE
+                 ENABLE_BROWSER_PANELS ENABLE_BROWSER_QT_LOOP)
 
-set(obs-browser_LIBRARIES
-	libobs
-	obs-frontend-api
-	)
+find_package(CEF REQUIRED)
```

Comment:
You should maybe add a fatal error when it's not found. rather than letting CMake continue.

Reply:
Added.